moonraker/components/machine.py

```diff
@@ -1155,6 +1156,113 @@ def _process_raw_properties(
+        self.server.add_warning(
+            "[machine]: Supervisord manager can not process SHUTDOWN."
+            "Please try KILL container or stop Supervisord via ssh terminal."
+        )
+        return
+
+    async def reboot(self) -> None:
+        self.server.add_warning(
+            "[machine]: Supervisord manager can not process REBOOT."
+            "Please try KILL container or stop Supervisord via ssh terminal."
+        )
+        return
+
+    async def do_service_action(
+        self, action: str, service_name: str
+    ) -> None:
+        # slow reaction for supervisord, timeout set to 6.0
+        await self._exec_command(
+            f"supervisorctl {self.spv_conf}"
+            f"{action} {service_name}",
+            timeout=6.
+        )
+
+    async def check_virt_status(self) -> Dict[str, Any]:
+        virt_id = virt_type = "none"
+        if (
+            os.path.exists("/.dockerenv") or
+            os.path.exists("/.dockerinit")
+        ):
+            virt_id = "docker"
+            virt_type = "docker"
+        return {
+            'virt_type': virt_type,
+            'virt_identifier': virt_id
+        }
+
+    async def _exec_command(
+        self, command: str, tries: int = 1, timeout=2.
+    ) -> str:
+        return await self.shell_cmd.exec_cmd(
+            command, proc_input=None, log_complete=False, retries=tries,
+            timeout=timeout
+        )
+
+    async def _update_service_status(self,
+                                     sequence: int,
+                                     notify: bool = True
+                                     ) -> None:
+        if sequence % 2:
+            # Update every other sequence
+            return
+        svcs = list(self.available_services.keys())
+        try:
+            # slow reaction for supervisord, timeout set to 6.0
+            resp = await self.svc_cmd.run_with_response(
+                log_complete=False, timeout=6.
+            )
+            resp_l = resp.strip().split("\n")  # drop lengend
+            for svc, state in zip(svcs, resp_l):
+                sub_state = state.split()[1].lower()
+                new_state: Dict[str, str] = {
+                    'active_state': "active",
+                    'sub_state': sub_state
+                }
+                if self.available_services[svc] != new_state:
+                    self.available_services[svc] = new_state
+                    if notify:
+                        self.server.send_event(
+                            "machine:service_state_changed",
+                            {svc: new_state})
+        except Exception:
+            logging.exception("Error processing service state update")
+
+    # service files is defined since docker build.
+    # not needed to implete SV.
+    async def extract_service_info(
+        self,
+        service: str,
+        pid: int,
+        properties: List[str],
+        raw: bool = False
+    ) -> Dict[str, Any]:
+        return {}
```

Comment:
Does `supervisord` provide a mechanism to retrieve information about a service?  If so it would be preferable to attempt to report what is possible.

Reply:
No, `supervisord` do not provide the information for running service (only running time and status). if neccessery information can be extracted from config file of `supervisord`.

Reply:
It isn't immediately necessary, we'll leave it as an empty dict for the time being.